packages/extension/src/view/popup/stateProviders/settings/settingsProvider.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies.
+ */
+import React, {
+  type PropsWithChildren,
+  useEffect,
+  useState,
+  useCallback,
+} from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Context from './context';
+import { SERVICE_WORKER_TABS_RELOAD_COMMAND } from '../../../../constants';
+
+const Provider = ({ children }: PropsWithChildren) => {
+  const [allowedNumberOfTabs, setAllowedNumberOfTabs] = useState<string | null>(
+    null
+  );
+  const [
+    allowedNumberOfTabsForSettingsDisplay,
+    setAllowedNumberOfTabsForSettingsDisplay,
+  ] = useState<string | null>(null);
+
+  const [settingsChanged, _setSettingsChanged] = useState<boolean>(false);
+
+  const [isUsingCDP, _setIsUsingCDP] = useState(false);
+  const [isUsingCDPForSettingsDisplay, setIsUsingCDPForSettingsDisplay] =
+    useState(false);
+
+  const intitialSync = useCallback(async () => {
+    const sessionStorage = await chrome.storage.session.get();
+    const currentSettings = await chrome.storage.sync.get();
+
+    if (Object.keys(sessionStorage).includes('pendingReload')) {
+      _setSettingsChanged(sessionStorage?.pendingReload);
+
+      if (Object.keys(sessionStorage).includes('allowedNumberOfTabs')) {
+        setAllowedNumberOfTabsForSettingsDisplay(
+          sessionStorage.allowedNumberOfTabs
+        );
+      } else {
+        setAllowedNumberOfTabsForSettingsDisplay(
+          currentSettings.allowedNumberOfTabs
+        );
+      }
+
+      if (Object.keys(sessionStorage).includes('isUsingCDP')) {
+        setIsUsingCDPForSettingsDisplay(sessionStorage.isUsingCDP);
+      } else {
+        setIsUsingCDPForSettingsDisplay(currentSettings.isUsingCDP);
+      }
+    } else {
+      setAllowedNumberOfTabsForSettingsDisplay(
+        currentSettings.allowedNumberOfTabs
+      );
+      setIsUsingCDPForSettingsDisplay(currentSettings.isUsingCDP);
+    }
+
+    if (Object.keys(currentSettings).includes('allowedNumberOfTabs')) {
+      setAllowedNumberOfTabs(currentSettings.allowedNumberOfTabs);
+    }
+
+    if (Object.keys(currentSettings).includes('isUsingCDP')) {
+      _setIsUsingCDP(currentSettings.isUsingCDP);
+    }
+  }, []);
+
+  const setUsingCDP = useCallback((newValue: boolean) => {
+    setIsUsingCDPForSettingsDisplay(newValue);
+    chrome.storage.session.set({
```

Comment:
Isn't this a promise? I think this should be "awaited".

Reply:
Resolved in [4ee3a41](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/519/commits/4ee3a417178866e7763cc9638d18ccd03c5a8f5c)